src/views/List.jsx

```diff
@@ -1,16 +1,48 @@
 import { ListItem } from '../components';
+import { useState } from 'react';
 
 /** List component that displays items in a user's shopping cart  */
 export function List({ data }) {
+	const [filteredList, setFilteredList] = useState(null);
+	const [filterInput, setFilterInput] = useState(null);
+
+	/* Use handler to change the state of filterInput  and convert all items to lowercase to facilitate a more thorough search */
+	const handleInput = (event) => {
+		const value = event.target.value.toLowerCase();
+		setFilterInput(value);
+		setFilteredList(
+			data.filter((item) => {
+				const name = item.name.toLowerCase();
+				return name.includes(value);
```

Comment:
One way you can filter item names without including the outlying `id` object that's created with a new list, is to use bracket notation to check if the element has a `name` property:

> item["name"]